src/app/hooks/useScrollDepth/index.jsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState, useContext } from 'react';
+import { OptimizelyContext } from '@optimizely/react-sdk';
+import { RequestContext } from '#contexts/RequestContext';
+import useOptimizelyVariation from '#hooks/useOptimizelyVariation';
+
+const getScrollDepth = () =>
+  Math.floor(
+    (document.documentElement.scrollTop /
+      (document.documentElement.scrollHeight -
+        document.documentElement.clientHeight)) *
+      100,
+  );
+
+const useScrollDepth = () => {
+  const { isAmp } = useContext(RequestContext);
+  const { optimizely } = useContext(OptimizelyContext);
+  const [scrollDepth, setScrollDepth] = useState(0);
+  const [scrollTwentyFive, setScrollTwentyFive] = useState(false);
+  const [scrollFifty, setScrollFifty] = useState(false);
+  const [scrollSeventyFive, setScrollSeventyFive] = useState(false);
+  const [scrollHundred, setScrollHundred] = useState(false);
+
+  const promoVariation = useOptimizelyVariation();
+  const hasVariationKey = promoVariation !== null;
+
+  const sendScrollEvents = hasVariationKey && !isAmp;
+
+  useEffect(() => {
+    if (scrollDepth >= 25 && !scrollTwentyFive && sendScrollEvents) {
```

Comment:
Not sure if it makes it any cleaner, but would something like `if(!sendScrollEvents) return;` work here, rather than check for `sendScrollEvents` in each condition below?

Reply:
Done 👍